test/integration/gardenlet/backupbucket/backupbucket_test.go

```diff
@@ -0,0 +1,323 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backupbucket_test
+
+import (
+	"fmt"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/gardenlet/controller/backupbucket"
+	"github.com/gardener/gardener/pkg/utils/test"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("BackupBucket controller tests", func() {
+	var (
+		gardenSecret                      *corev1.Secret
+		backupBucket                      *gardencorev1beta1.BackupBucket
+		seedGeneratedSecret               *corev1.Secret
+		gardenGeneratedSecret             *corev1.Secret
+		extensionSecret                   *corev1.Secret
+		extensionBackupBucket             *extensionsv1alpha1.BackupBucket
+		expectedExtensionBackupBucketSpec extensionsv1alpha1.BackupBucketSpec
+		providerStatus                    = &runtime.RawExtension{Raw: []byte(`{"foo":"bar"}`)}
+
+		backupBucketReady = func(makeReady bool) {
```

Comment:
This name is a bit confusing since it takes an argument that's called makeReady which can be false or true. You should probably rename the function to `reconcileExtensionsBackupBucket`

Reply:
We're making it ready and not ready. That's why 😄